test/acceptance/features/bindAppToService.feature

```diff
@@ -586,3 +585,86 @@ Feature: Bind an application to a service
+        When Invalid Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-without-spec
+            """
+        Then Error message "spec: Required value" is thrown
+        And Service Binding "binding-request-without-spec" is not persistent in the cluster
+
+        @negative1
+        Scenario: Service Binding with empty spec is not allowed in the cluster
+        When Invalid Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-empty-spec
+            spec:
+            """
+        Then Error message "invalid: spec: Invalid value: \"null\"" is thrown
+        And Service Binding "binding-request-empty-spec" is not persistent in the cluster
+
+        @negative1
+        Scenario: Service Binding with empty services is not allowed in the cluster
+        When Invalid Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-empty-services
+            spec:
+                services:
+            """
+        Then Error message "invalid: spec.services: Invalid value: \"null\"" is thrown
+        And Service Binding "binding-request-empty-services" is not persistent in the cluster
+
+        @negative1
+        Scenario: Service Binding without gvk of services is not allowed in the cluster
+        When Invalid Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: binding-request-without-gvk
+            spec:
+                services:
+                -   name: backend-demo
+            """
+        Then Error message "spec.services.group: Required value" is thrown
+        And Error message "spec.services.kind: Required value" is thrown
+        And Error message "spec.services.version: Required value" is thrown
+        And Service Binding "binding-request-without-gvk" is not persistent in the cluster
+
+        @negative1
+        Scenario: Configuring exsting service binding with invalid spec configuration is not allowed
+        Given Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: persisting-binding-request
+            spec:
+                services:
+                -   group: service.example.com
+                    version: v1
+                    kind: Backserv
+                    name: demo-backserv-cr-2
+            """
+        Then Get "{.metadata.resourceVersion}" of "servicebinding" "persisting-binding-request"
```

Comment:
what is the purpose of this step?

Reply:
This step will save the resourceVersion of first CR in context and we'll later verify if the resourceVersion is same or changed to validate the unchanged state of service binding.

Reply:
This smells like abstraction leakage, i.e. implementation details. User should not know anything about that. The only thing what users sees is an error after trying to update the service binding. Think about the scenario as list of instruction we give user/tester so he is able to verify the scenario. Whatever we do in the line 732 should gather enough information (for example this step could be a part of that). so that we can perform later steps.

Reply:
Redesigned my approach PTAL.